api/dao/containerstorage.py

```diff
@@ -54,8 +55,25 @@ def _update_el(self, _id, payload):
                 _id = bson.objectid.ObjectId(_id)
             except bson.errors.InvalidId as e:
                 raise APIStorageException(e.message)
+        if recursive:
+            self._propagate_changes(_id, update)
         return self.dbc.update_one({'_id': _id}, update)
 
+    def _propagate_changes(self, _id, update):
+        """
+        Propagates changes down the heirarchy tree when a PUT is marked as recursive.
+        """
+
+        if self.cont_name == "projects":
+            session_ids = [s['_id'] for s in config.db.sessions.find({'project': _id}, [])]
+            config.db.sessions.update_many(
+                {'project': _id}, update)
+            config.db.acquisitions.update_many(
+                {'session': {'$in': session_ids}}, update)
+        elif self.cont_name == "sessions":
+            config.db.acquisitions.update_many(
+                {'session': _id}, update)
```

Comment:
Add an `else` that throws an exception - right now, `_propagate_changes` will happily do nothing if `cont_name` is not in the list.